claat/parser/md/parse.go

```diff
@@ -127,6 +127,13 @@ type stackItem struct {
 	cur *html.Node
 }
 
+func newDocState() *docState {
```

Comment:
Method duplicate given different parser. Might be a good idea of having another pkg with all these constructors by themselves.

Reply:
Worth looking into the the future but IMO out of scope for this change.

Reply:
If you find yourself making more duplicate constructors, then yeah this should be how is done. I agree this is small enough but the pattern might emerge eventually.

I did something similar for the proto constructors here https://github.com/googlecodelabs/tools/blob/dev/claat/proto/constructor/helpers.go

Reply:
update: seem like you're already doing this with the `NewCodelab` constructor in `types/meta`